perl/README.md

```diff
@@ -0,0 +1,1028 @@
+
+4. **Reserved Delimiters**: The reserved delimiters described in points 1-3 (`;`, `/`, and `|`) will be stripped and replaced with a comma if found in source inputs to the Bystro Annotation Database.
+
+## What Information Can Bystro Annotator Output?
+
+Bystro Annotator is a general-purpose data curation and labeling engine for genetic data, and has no restrictions on the types of annotations/feature labels it can output. Theoretically it can even support binary data, such as images.
+
+## Variant Representations
+
+Bystro's variant representation deviates slightly from the standard VCF format in the name of simplicity. In particular, it drops the rule that the alternate allele must be ACTG. Dropping this single restriction allows us to represent deletions as occuring at the actual first deleted base, rather than the base before, as in the VCF format. This has a number of knock on benefits:
+
+- The `inputRef` (reference base) in Bystro's annotation outputs is always exactly 1 base long
+
+- The `pos' (position) in Bystro's annotation outputs is always the first affected base, except in the case of insertions, where it is the base before the insertion, since the insertion by definition is between two reference bases
+
+- It is possible to represent all multiallelic site using a single reference base, a single position, and a list of alleles
+
+The Bystro Genotype Dosage Matrix, is a columnar dataset, generated for every collection of VCFs submitted. Its first column is the `locus`, which is `chr:pos:ref:alt`. Every column after that is labeled by the sample name, and contains a -1 for missing genotypes, 0 for reference, 1 for a single alternate allele, 2 for 2 alternate alleles (homozygous in a diploid organism), and so on. It can be used for many things, such as to calculate polygenic risk scores.
+
+### Comparing the VCF and Bystro Variant Representations
+
+Below we'll demonstrate the Bystro Annotation and Dosage Matrix outputs, using some examples. You may run these examples, if you have Bystro installed, or at least the Bystro VCF Preprocessor. The Bystro VCF Preprocessor, which is a tool that takes a VCF file and outputs a Bystro-annotator compatible TSV file, which is then used by the Perl Bystro Annotator to add annotations to the VCF file from the Bystro Annotation Database. You can install it by running `go install github.com/bystrogenomics/bystro-vcf@2.2.3`.
+
+Please note that we are not showing the full Bystro Annotator outputs, which are far too extensive to easily display here.
+
+```
+cat ~/bystro/perl/example_vcf.tsv | bystro-vcf --keepId --emptyField "NA" --keepPos
+```
+
+If you also want to output the **Bystro Genotype Dosage Matrix**, at a small performance hit, you can run:
```

Comment:
Not sure it's necessary to mention a performance hit if it's small? I would assume asking something to do more would take longer but if it's 'small' then it doesn't seem worth drawing attention to it.

Reply:
The performance comparisons were without this, so I need to make a reference to it.